statsmodels/tsa/statespace/kalman_smoother.py

```diff
@@ -30,63 +31,117 @@
 ))
 
 
-def _kalman_smooth(model, kfilter, smoother_output):
-    # Create storage
-    scaled_smoothed_estimator = None
-    scaled_smoothed_estimator_cov = None
-    smoothing_error = None
-    smoothed_state = None
-    smoothed_state_cov = None
-    smoothed_state_disturbance = None
-    smoothed_state_disturbance_cov = None
-    smoothed_measurement_disturbance = None
-    smoothed_measurement_disturbance_cov = None
-
-    # Intermediate values
-    tmp_L = np.zeros((model.k_states, model.k_states, model.nobs),
-                     dtype=kfilter.dtype)
-
-    if smoother_output & (SMOOTHER_STATE | SMOOTHER_DISTURBANCE):
-        scaled_smoothed_estimator = np.zeros((model.k_states, model.nobs + 1),
-                                             dtype=kfilter.dtype)
-        smoothing_error = np.zeros((model.k_endog, model.nobs),
-                                   dtype=kfilter.dtype)
-    if smoother_output & (SMOOTHER_STATE_COV | SMOOTHER_DISTURBANCE_COV):
-        scaled_smoothed_estimator_cov = (
-            np.zeros((model.k_states, model.k_states, model.nobs + 1),
-                     dtype=kfilter.dtype)
-        )
+class _kalman_smoother(object):
```

Comment:
Sounds fine to me.
I'm mainly commenting on pep-8, Putting the internal code with leading underline and the same name is quite common. (unless it needs additional qualifiers in the name)